### ISSUE TITLE
Add music source separation (Mel-Band RoFormer + Open-Unmix)

### ISSUE THREAD

**Issue by ivan-digital** [ivan-digital/qwen3-asr-swift#94]
## Summary

Add music source separation to split a mixed audio track into 4 stems: **vocals, drums, bass, other**. Two backends for different use cases:

- **Mel-Band RoFormer** — SOTA quality (~11.4 dB SDR), offline, for best vocal/instrument isolation
- **Open-Unmix** — real-time streaming (~6.2 dB SDR), low-latency, for live applications

## Why both?

| | Mel-Band RoFormer | Open-Unmix |
|---|---|---|
| **Use case** | Studio quality, post-processing | Live monitoring, DJ, karaoke |
| **Quality (SDR)** | ~11.4 dB (near-perfect) | ~6.2 dB (good) |
| **Latency** | Offline (seconds) | ~23ms streaming |
| **Params** | ~93M (standard), smaller variants exist | ~8.6M |
| **Weights** | Many on HuggingFace (MIT/CC) | umxhq (MIT), umxl (CC BY-NC-SA) |
| **Architecture** | STFT + mel-band split + RoPE Transformer | STFT + BiLSTM + magnitude mask |
| **Code reuse** | RoPE, attention, STFT (all in-repo) | BiLSTM, STFT (all in-repo) |

### Architecture

```
Stereo 44.1kHz input
    -> STFT (2048 window)
    -> Mel-band split (divide spectrogram into overlapping subbands)
    -> Per-band: Linear projection -> RoPE Transformer (12 blocks, 8 heads)
    -> Band merge (reconstruct full spectrogram)
    -> Complex mask estimation
    -> Apply mask + iSTFT
    -> 4 stereo stems
```

### Pretrained weights (no training needed)

| Weights | Source | Quality | License |
|---|---|---|---|
| model_mel_band_roformer_ep_3005_sdr_11.4360 | viperx | 11.4 dB | Open |
| MelBandRoformer.ckpt | KimberleyJSN | ~10.8 dB | Open |
| Mel-Band-Roformer-small | pcunwa | ~9.5 dB | Open |
| 4-stem model | ZFTurbo | 8.9 dB | Open |

### Implementation

**Module**: `Sources/SourceSeparation/`
- `MelBandRoFormer.swift` — Transformer blocks with RoPE, band-split/merge
- `SourceSeparation.swift` — High-level API
- `WeightLoading.swift` — Checkpoint to MLX weight mapping

**Conversion**: `scripts/convert_mel_roformer.py`
- Load .ckpt weights from community checkpoints
- Export to safetensors (optionally quantize to 4-bit)

**Existing code to reuse**:
- RoPE — `Qwen3Common` (used across ASR/TTS/CosyVoice)
- Transformer attention — multiple modules
- vDSP STFT/iSTFT — `SpeechEnhancement/AudioProcessing.swift`

**New components needed**:
- Mel-band split/merge module (overlapping subband projection)
- Complex mask estimation and application
- 44.1kHz stereo audio path (current pipeline is mono 16/24kHz)

### Architecture

```
Stereo 44.1kHz input (streaming, 1024-hop chunks)
    -> STFT (4096 window, 1024 hop)
    -> Magnitude spectrogram
    -> Linear + BatchNorm
    -> 3x BiLSTM (stateful, carry h/c across chunks)
    -> Linear -> Sigmoid mask
    -> Multiply with input magnitude
    -> iSTFT
    -> 4 stereo stems
```

One model per source (vocals, drums, bass, other).

### Implementation

- `OpenUnmixModel.swift` — BiLSTM + Linear, magnitude masking
- Streaming via LSTM state carry-across (same pattern as Silero VAD)
- Reference: [umx.cpp](https://github.com/sevagh/umx.cpp) (C++ streaming port with quantization)

**Existing code to reuse**:
- BiLSTM — `SpeechVAD/PyanNetModel.swift`
- Streaming state machine — `SpeechVAD/StreamingVADProcessor.swift`
- vDSP STFT — `SpeechEnhancement/AudioProcessing.swift`

## CLI

```bash
# Quality mode (Mel-Band RoFormer, default)
audio separate song.wav                              # all 4 stems
audio separate song.wav --stems vocals               # vocals only
audio separate song.wav --stems vocals,drums          # select stems
audio separate song.wav --output-dir stems/           # custom output dir
audio separate song.wav --model roformer-small        # faster, smaller model

# Real-time mode (Open-Unmix)
audio separate song.wav --realtime                    # streaming separation
```

### Quality metrics (SDR on MUSDB18-HQ test set)
- Per-stem SDR: vocals, drums, bass, other
- Compare against published baselines for both models
- Test quantized weights (4-bit RoFormer, uint8 UMX) vs full precision
- Evaluate with [museval](https://github.com/sigsep/sigsep-mus-eval)

### Performance (Apple Silicon)
- RTF on M1/M2/M3/M4 for both models
- Peak memory / RSS
- RoFormer: time per track (offline)
- Open-Unmix: latency per chunk (streaming)
- Model load time + first-inference warmup

### Perceptual
- A/B comparison: RoFormer vs Open-Unmix on same tracks
- Genre coverage: pop, rock, jazz, classical, electronic
- Failure mode analysis: where does each model break down?

## References

- [ZFTurbo/Music-Source-Separation-Training](https://github.com/ZFTurbo/Music-Source-Separation-Training)
- [lucidrains/BS-RoFormer](https://github.com/lucidrains/BS-RoFormer)
- [Mel-Band RoFormer paper](https://arxiv.org/abs/2310.01809)
- [sigsep/open-unmix-pytorch](https://github.com/sigsep/open-unmix-pytorch)
- [sevagh/umx.cpp](https://github.com/sevagh/umx.cpp)
- [MUSDB18-HQ dataset](https://sigsep.github.io/datasets/musdb.html)
- [museval](https://github.com/sigsep/sigsep-mus-eval)
